Keithley2612B_voltage_sweep.py

```diff
@@ -162,33 +183,39 @@ def sweep_operation(smu_id, steps_no, pattern, nplc, min, max, scan_rate):
         # Get the currents
         smu_id.write(f"printbuffer(1, {steps_no}, smua.nvbuffer1.readings)")
         current_string = smu_id.read()
-
+        print("Currents obtained (string form).")
     
         # Get the voltages
         smu_id.write(f"printbuffer(1, {steps_no}, smua.nvbuffer2.readings)")
         voltage_string = smu_id.read()
-
+        print("Voltages obtained (string form).")
     
         # Get the timestamps
         smu_id.write(f"printbuffer(1, {steps_no}, smua.nvbuffer1.timestamps)")
         timestamp_string = smu_id.read()
+        print("Timestamps obtained (string form).")
     
         current_string_array = current_string.split(',')
         voltage_string_array = voltage_string.split(',')
         timestamp_string_array = timestamp_string.split(',')
     
         currents = np.array(current_string_array, dtype=float)
+        print("Successful conversion of current string to array of floats.")
         voltages = np.array(voltage_string_array, dtype=float)
+        print("Successful conversion of voltage string to array of floats.")
         timestamps = np.array(timestamp_string_array, dtype=float)
+        print("Successful conversion of timestamp string to array of floats.")
 
         outputs[i] = [voltages, currents, timestamps]
+        print("Information pushed to index i = " + str(i) + " of outputs list.")
 
         smu_id.write ("smua.nvbuffer1.clear()")
         smu_id.write ("smua.nvbuffer2.clear()")
         smu_id.write ("smua.nvbuffer1.clearcache()")
         smu_id.write ("smua.nvbuffer2.clearcache()")
-    
-
+        print("Buffers cleared again.")
+    print(outputs)
```

Comment:
One of the issues that are not resolved yet is that although we can print the outputs, we are not storing the outputs anywhere when we use the 'START RUN' button. As you can see in the GUI file, I have not assigned any variable that stores the returned outputs. We will need those outputs to plot the JV curves. I am not sure if we can even return anything from the functions that we set as commands to a button.
I was thinking maybe we can a functionality to the scan function which directly writes the output to a csv then we can read that file to plot the curves. It sounds redundant though. Can you think of an alternative or should we just go with this?

Reply:
I think this depends on how we can work with matplotlib with Tkinter. If we can send the matplotlib axes object as a parameter, then maybe we can write a dedicated function/file just for plotting. 

The third answer here might help too: https://stackoverflow.com/questions/13099908/python-tkinter-return-value-from-function-used-in-command

But it leads back to the same issue. I'm not sure how to communicate with Matplotlib from Tkinter.